it/repositories/MongoJourneyAnswersRepositoryISpec.scala

```diff
@@ -16,123 +16,97 @@
-  private val id                 = "some_id"
-  private val someJourneyAnswers = JourneyAnswers(id, Json.obj("field" -> "value"), someOldTimestamp)
+  override val repository = new MongoJourneyAnswersRepository(mongoComponent, clock)
 
   override def beforeEach(): Unit = {
-    super.beforeEach()
-    await(removeAll())
+    clock.reset(now)
+    await(removeAll(repository.collection))
   }
 
-  "MongoJourneyAnswersRepository" when {
-    "initialised" must {
-      "include an TTL index for the `lastUpdated` field (where expiry is set through the app config)" in {
-        checkIndex(indexWithField("lastUpdated")) { indexModel =>
-          indexModel.getOptions.getExpireAfter(TimeUnit.DAYS) shouldBe 2
-        }
-      }
+  "upsertData" should {
+    "insert a new journey answers in in-progress status and calculate dates" in {
+      val result = (for {
+        _        <- repository.upsertData(mtditid, currTaxYear, businessId, JourneyName.Income, Json.obj("field" -> "value"))
+        inserted <- repository.get(mtditid, currTaxYear, businessId, JourneyName.Income)
+      } yield inserted.value).futureValue
+
+      val expectedExpireAt = ExpireAtCalculator.calculateExpireAt(now)
+      result shouldBe JourneyAnswers(
+        mtditid,
+        businessId,
+        currTaxYear,
+        JourneyName.Income,
+        InProgress,
+        Json.obj("field" -> "value"),
+        expectedExpireAt,
+        now,
+        now)
     }
-    "setting journey answers" when {
-      "no journey answers exist for the supplied id" must {
-        "store them and update the `lastUpdated` field to now" in {
-          repository.set(someJourneyAnswers).futureValue shouldBe ()
-
-          withClue("lastUpdated was not updated to now") {
-            lastUpdated(someJourneyAnswers.id) shouldBe Some(now)
-          }
-        }
-      }
-      "journey answers exist for the supplied id" must {
-        "update them " in {
-          await(repository.set(someJourneyAnswers)) shouldBe ()
-
-          val updatedData           = Json.obj("field" -> "updatedValue")
-          val updatedJourneyAnswers = someJourneyAnswers.copy(data = updatedData)
-
-          await(repository.set(updatedJourneyAnswers)) shouldBe ()
-
-          repository.get(id).futureValue.map(_.data) shouldBe Some(updatedData)
 
-          withClue("lastUpdated was not updated to now") {
-            lastUpdated(updatedJourneyAnswers.id) shouldBe Some(now)
-          }
-        }
-      }
+    "update already existing answers (values, updateAt)" in {
+      val result = (for {
+        _ <- repository.upsertData(mtditid, currTaxYear, businessId, JourneyName.Income, Json.obj("field" -> "value"))
+        _ = clock.advanceBy(1.day)
+        _       <- repository.upsertData(mtditid, currTaxYear, businessId, JourneyName.Income, Json.obj("field" -> "updated"))
+        updated <- repository.get(mtditid, currTaxYear, businessId, JourneyName.Income)
+      } yield updated.value).futureValue
```

Comment:
Does this add test value to check the document has actually been modified?

```scala
      val result = (for {
        _ <- repository.upsertData(mtditid, currTaxYear, businessId, JourneyName.Income, Json.obj("field" -> "value"))
        _ = clock.advanceBy(1.day)
        updateResult <- repository.upsertData(mtditid, currTaxYear, businessId, JourneyName.Income, Json.obj("field" -> "updated"))
        updated      <- repository.get(mtditid, currTaxYear, businessId, JourneyName.Income)
      } yield {
        updateResult.getModifiedCount shouldBe 1
        updated.value
      }).futureValue
```

Reply:
I can add it to be on the safe side